Reducing test set size and adding a script to crop dicom images.

tdraebing opened the pull request:
Replaced large test images with smaller ones to decrease repo size.

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
I decreased the size of all three data sets by reducing the number of images (cropping in z-dimension). For the data sets LIDC-IDRI-0001 and LIDC-IDRI-0003 I chose the regions that contain a big mass inside the lung, which are probably tumors. LIDC-IDRI-0002 does not contain such a mass. Thus I chose a range of slices that contained nodules as stated by the *.xml file. (Disclaimer: I am not a medical professional. As a biologist I only have very limited knowledge about diagnosing lung cancer on CT-images). The *.xml files were removed from the data set for now. The data sets are in the same location as the original data sets. The names changed to be more declarative and now represent the depth of the slices in mm instead of a non-consecutive id.
Additionally a script was added to `prediction/src/preprocess/crop_dicom.py` that allows to crop dicom series in x-, y- and z-dimensions for future use.
Existing tests were updated accordingly.

## Reference to official issue
<!--- If fixing a bug, there should be an existing issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
Issue #38 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If adding a new feature or making improvements not already reflected in an official issue, please reference the relevant sections of the design doc -->
The datasets were quite large. Smaller datasets would substentially decrease the repo size. The created script might be useful for preprocessing in prediction tasks.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Unit tests for the new script were created to test whether the expected output is created. Existing tests were updated to appropriately deal with the smaller test sets (One test relied on a *.xml-file being in the folder). The other tests resulted in no errors. Smaller datasets can be opened using an external DICOM-Viewer (RadiAnt).

## Screenshots (if appropriate):


## CLA
- [x] I have signed the CLA; if other committers are in the commit history, they have signed the CLA as well

reubano commented:
Thanks @tdraebing! Nice job with this. One question initially: would you mind using [Google style formatting](https://github.com/concept-to-clinic/concept-to-clinic/blob/c484acf1af419235ee98889332c1c890a35a140c/prediction/src/preprocess/load_dicom.py#L39-L43) for the docblocks?

tdraebing commented:
I updated the pull request. The doc string now follows Google style formatting. Also the full test set was included again next to the cropped dataset. I updated the docker-configuration to load the small dataset into the `/images`-folder when running tests.

reubano commented on prediction/src/preprocess/crop_dicom.py:
```diff
@@ -0,0 +1,50 @@
+
+        The given DICOM series contains images with the resolution of 512x512 pixels and 133 images
+        starting at a depth of -10 mm to a depth of -340 mm. Using begin = [0, 0, -10] and end = [512, 512, -340] would
+        give you back the full series. Using begin = [100, 200, [0, 0, -160] and end = [120, 220, -170] would give you a
```
`[100, 200, [0, 0, -160]` is this a typo?

reubano commented on prediction/src/tests/test_crop_dicom.py:
```diff
@@ -0,0 +1,35 @@
+    assert np.asarray(list(filter(lambda x: x.SliceLocation == -102.5, cropped_series))[0].pixel_array)[0][0] == \
+        np.asarray(list(filter(lambda x: x.SliceLocation == -102.5, uncropped_series))[0].pixel_array)[100][100]
+    assert np.asarray(list(filter(lambda x: x.SliceLocation == -102.5, cropped_series))[0].pixel_array)[19][19] == \
+        np.asarray(list(filter(lambda x: x.SliceLocation == -102.5, uncropped_series))[0].pixel_array)[119][119]
```
I think we can replace `list(filter(lamba x...`'s here with list comprehensions, e.g., `[x for x in cropped_series if x.SliceLocation == -102.5]`

reubano commented:
Made a couple comments... afterwards, a rebase + squash should be all that's needed. Thanks!

tdraebing commented:
Thanks for reviewing! Should be fixed now.

reubano commented:
Quick question, does the small test data set use the actual images or the `git lfs` ones? It should be the actual images, but I just ran a test and noticed the files on my computer are still linked via `lfs`.

tdraebing commented:
I am not very familiar with `lfs`, but as I understand it, right now `lfs` is scanning for all `*.dcm`-files in the repo-folder as specified in the `.gitattributes`-file. Thus the small dataset  would also be handled by `lfs`. `git lfs ls-files` indeed shows that the small datasets are managed by `lfs`.
I am not completely sure how to fix it correctly without doing a mistake. I guess I would have to follow this guide: https://help.github.com/articles/removing-files-from-git-large-file-storage/ and change the entry in `.gitattributes` to a more specific path. But as I understand it, this will completely remove the files from the repo. Is changing the `.gitattributes`-file and committing with the image files maybe enough to replace the `lfs`-pointers in the repo with the proper files? Could anybody comment on that?

reubano commented:
Ok... maybe a fix can bundled in with #59? @lamby @pjbull
